apps/website/src/components/courses/CourseDirectory.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  useState,
+  ChangeEvent,
+  FC,
+  useCallback,
+} from 'react';
+import { AxiosError } from 'axios';
+import clsx from 'clsx';
+import {
+  Section,
+  ProgressDots,
+  ErrorSection,
+} from '@bluedot/ui';
+import { H3, P } from '@bluedot/ui/src/Text';
+import { FaFilter } from 'react-icons/fa6';
+import type { CoursesRequestBody, GetCoursesResponse } from '../../pages/api/courses';
+import { CourseSearchCard } from './CourseSearchCard';
+
+type Option = {
+  value: string;
+  label: string;
+};
+
+const CADENCE_OPTIONS: Option[] = [
+  { value: 'MOOC', label: 'Self-paced' },
+  { value: 'Daily', label: 'Daily' },
+  { value: 'Weekly', label: 'Weekly' },
+];
+const LEVEL_OPTIONS: Option[] = [
+  { value: 'Beginner', label: 'Beginner' },
+  { value: 'Intermediate', label: 'Intermediate' },
+  { value: 'Advanced', label: 'Advanced' },
+];
+
+export type CourseDirectoryProps = {
+  displayData: GetCoursesResponse | undefined;
+  displayLoading: boolean;
+  displayError: AxiosError<unknown, unknown> | null | undefined;
+  noResults: boolean;
+  refetch: (filters: CoursesRequestBody) => void;
+};
+
+const CourseDirectory: FC<CourseDirectoryProps> = ({
+  displayData,
+  displayLoading,
+  displayError,
+  noResults,
+  refetch,
+}) => {
+  const [selectedCadences, setSelectedCadences] = useState<string[]>(CADENCE_OPTIONS.map((c) => c.value));
+  const [selectedLevels, setSelectedLevels] = useState<string[]>(LEVEL_OPTIONS.map((l) => l.value));
+  const [filtersOpenMobile, setFiltersOpenMobile] = useState(false);
+
+  const handleCadenceChanged = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = event.target;
+    const newSelectedCadences = checked ? [...selectedCadences, value] : selectedCadences.filter((l) => l !== value);
+    setSelectedCadences(newSelectedCadences);
+    refetch({ cadence: newSelectedCadences, level: selectedLevels });
+  };
+
+  const handleLevelChanged = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = event.target;
+    const newSelectedLevels = checked ? [...selectedLevels, value] : selectedLevels.filter((l) => l !== value);
+    setSelectedLevels(newSelectedLevels);
+    refetch({ cadence: selectedCadences, level: newSelectedLevels });
+  };
```

Comment:
There is some duplication here but with only two filter types it's simpler overall to do it like this